packages/replay-next/src/utils/range-list.ts

```diff
@@ -0,0 +1,96 @@
+export interface Range {
+  begin: bigint;
+  end: bigint;
+}
+
+function min(b1: bigint, b2: bigint) {
+  return b1 < b2 ? b1 : b2;
+}
+
+function max(b1: bigint, b2: bigint) {
+  return b1 > b2 ? b1 : b2;
+}
+
+export function isInRange(b: bigint, range: Range) {
+  return b >= range.begin && b < range.end;
+}
+
+export function rangesOverlap(range1: Range, range2: Range) {
+  return range1.end > range2.begin && range1.begin < range2.end;
+}
+
+export function rangeContains(range1: Range, range2: Range) {
+  return range1.begin <= range2.begin && range1.end >= range2.end;
+}
+
+export function getMissingRanges(requested: Range, cached: Range[]): Range[] {
```

Comment:
Functions like this should have associated unit tests– particularly to cover edge cases.